addons/mail/controllers/thread.py

```diff
@@ -101,20 +101,12 @@ def mail_message_post(self, thread_model, thread_id, post_data, context=None, sp
                 'last_used': datetime.now(),
                 'ids': canned_response_ids,
             })
-        thread = request.env[thread_model].with_context(active_test=False).search([("id", "=", thread_id)])
-        thread = thread.with_context(active_test=True)
+        thread = request.env[thread_model]._get_thread_with_access(
+            thread_model, thread_id, kwargs.get("token"), kwargs.get("hash"), kwargs.get("pid")
```

Comment:
`mail` has normally no "information" about these `token`, `hash`, `pid`, is it possible at this step to just pass all `kwargs`?

Reply:
Yes, Done